### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by KrystianMarek**
#### What's this change do?
- build artifacts are located in ./build folder
- parts of the python code were rewritten for better readability
- better logging was introduced
- pytest was introduced
- scale tests against BigIP were introduced
- process of building payloads was broken down into 3 distinct steps
- scripts folder was removed
- old CI tools were removed
- support for new CI was added
- readme.rst was added in tests folder, to describe how the test could be run
- files were renamed in payload_templates folder
- missing readme.rst  documenting the payload templates was updated
- test checking for license header in *.py files was added

#### Where should the reviewer start?
./src
./tests

### CODE REVIEW

File: payload_templates/include_defaults.ptmpl [hunk 1]
```diff
@@ -0,0 +1,308 @@
+{
```
Comment (lukesiler) [inline payload_templates/include_defaults.ptmpl]: Should be .json files so as not to confuse - it is just data.

Comment (KrystianMarek) [inline payload_templates/include_defaults.ptmpl]: They are not the final payloads, final payloads will be constructed during test run.
One should know that it is not possible to take a file from payload_templates/ and post it to BIP.

**Comment by KrystianMarek** [comment]
@lukesiler 
1) readme added
https://github.com/F5Networks/f5-application-services-integration-iApp/tree/cleanup/tests

2) I do not have access to that documentation

3) Yes, the project will be now built with:
mgmt_build.py
parameters are mostly the same as for build.py

4) file is back
https://github.com/F5Networks/f5-application-services-integration-iApp/blob/cleanup/payload_templates/README.rst

5) we have an epic for this:
https://jira.pdsea.f5net.com/browse/CDO-833

you would be interested in:
https://jira.pdsea.f5net.com/browse/CDO-841
https://jira.pdsea.f5net.com/browse/CDO-842
https://jira.pdsea.f5net.com/browse/CDO-704

**Comment by lukesiler** [comment]
#2 - please reach out to Joe Pruitt to determine the source of this content.  I thought it was all published from content in the github repo.

#3 - if there is no change then renaming seems confusing

#4 - please add "scale" to verification criteria for the test cases that are part of scale run.  Can you also please comment on why some are not compatible with scale?

File: payload_templates/include_defaults.ptmpl (hunk 1 above)
Comment (lukesiler) [inline payload_templates/include_defaults.ptmpl]: It is JSON data.  The manner in which the data is used, processed, combined with other data, etc. are all much broader topics.

Take the feedback - you're making it more confusing by inventing a file type.  Instead, just write clear docs on how the json data is currently used/abused so others don't have to do the forensics that you did.

File: tests/README.rst [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+   -->
+
+Testing
+=========================================================
+
+This folder contains pytest compatible tests.
+
+Run
+
+pytest --policy_host=IP_ADDRESS_OF_POLICY_HOST_SERVER --host=IP_ADDRESS_OF_BIG_IP --scale_size=NUMBER_OF_IAPPS_DEPLOYED_DURING_SCALE_RUN
```
Comment (lukesiler) [inline tests/README.rst]: please explain what is "policy host server"?

**Comment by KrystianMarek** [comment]
@lukesiler 
2) I will
3) The change in the source code is substantial
https://github.com/F5Networks/f5-application-services-integration-iApp/blob/develop/build.py
https://github.com/F5Networks/f5-application-services-integration-iApp/blob/cleanup/mgmt_build.py
I kept most of the command lines parameters and their functions

4) Scale tests are part of the pytest run
Pytest will run all unit and functional tests in 
https://github.com/F5Networks/f5-application-services-integration-iApp/tree/cleanup/tests
That were not deliberately skipped.

File: payload_templates/include_defaults.ptmpl (hunk 1 above)
Comment (KrystianMarek) [inline payload_templates/include_defaults.ptmpl]: It looks like a JSON, but it should not be treated as such. This is a 'JSON template' that uses a custom templating mechanism.

For example marker:
https://github.com/F5Networks/f5-application-services-integration-iApp/blob/cleanup/payload_templates/test_pools.ptmpl#L102
will be replaced by
https://github.com/F5Networks/f5-application-services-integration-iApp/blob/cleanup/src/appservices/PayloadGenerator.py#L90
during the configuration/Application Service payload build.

This is not my idea, this is how it was originally implemented, I just cleaned it up a little.
https://github.com/F5Networks/f5-application-services-integration-iApp/blob/master/test/run_tests.py#L62

File: tests/README.rst (hunk 2 above)
Comment (KrystianMarek) [inline tests/README.rst]: Please notice the updated version
https://github.com/F5Networks/f5-application-services-integration-iApp/tree/cleanup/tests

**Pull request merged**